src/bidiMapper/utils/log.ts

```diff
@@ -16,17 +16,27 @@
 export function log(type: string): (...message: any[]) => void {
   return (...messages: any[]) => {
     // If run in browser, add debug message to the page.
-    if (typeof window !== 'undefined' && window?.document?.documentElement) {
+    if (globalThis?.document?.documentElement) {
```

Comment:
```suggestion
    if (globalThis.document?.documentElement) {
```

I guess the first `?.` is unnecessary, since `globalThis` can't be null. My bad :)